plugin/pkg/admission/resourcequota/config_test.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourcequota
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+func TestLoadConfiguration(t *testing.T) {
+	// create a place holder file to hold per test config
+	configFile, err := ioutil.TempFile("", "admission-config")
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+	configFileName := configFile.Name()
+	defer os.Remove(configFileName)
+
+	if err = configFile.Close(); err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+	// individual test scenarios
+	testCases := []struct {
+		CaseName   string
+		ConfigBody string
+		ExpecteErr bool
+	}{
+		// valid configuration
+		{
+			CaseName: "valid configuration",
+			ConfigBody: `
+apiVersion: resourcequota.admission.k8s.io/v1alpha1
+kind: Configuration
+metadata:
+  name: quota-config
+limitedResources:
+  - resource: persistentvolumeclaims
+    matchContains:
+    - requests.storage
+  - resource: services
+    matchContains:
+      - loadbalancers
+      - nodeports
+  - resource: pods
+    matchContains:
+      - pods
+      - cpu
+`,
+			ExpecteErr: false,
+		},
+
+		// invalid configuration kind
+		{
+			CaseName: "invalid configuration kind",
+			ConfigBody: `
+apiVersion: resourcequota.admission.k8s.io/v1alpha1
+kind: invalid-Configuration
+metadata:
+  name: quota-config
+limitedResources:
+  - resource: persistentvolumeclaims
+    matchContains:
+    - requests.storage
+  - resource: services
+    matchContains:
+      - loadbalancers
+      - nodeports
+  - resource: pods
+    matchContains:
+      - pods
+      - cpu
+`,
+			ExpecteErr: true,
+		},
+
+		// invalid configuration
+		{
+			CaseName: "invalid configuration",
+			ConfigBody: `
+apiVersion: resourcequota.admission.k8s.io/v1alpha1
+kind: invalid-Configuration
+metadata:
+  name: quota-config
+limitedResources:
+  - resource: persistentvolumeclaims
+    unknown:
```

Comment:
does this fail because of the `unknown` or because of the invalid kind? I think the later.

Reply:
I will fix it, because of invalid kind

Reply:
Done.